File: src/common/com.smi.microservices.common/src/main/java/org/smi/common/messages/ExtractMessage.java

package org.smi.common.messages;

import java.util.UUID;

import org.smi.common.messageSerialization.JsonDeserializerWithOptions.FieldRequired;

public abstract class ExtractMessage implements IMessage {

	/**
	 * UUID for this run of the image extraction process
	 */
	@FieldRequired
	public UUID ExtractionJobIdentifier;

	/**
	 * Project number for reference and base of extracted files
	 */
	@FieldRequired
	public String ProjectNumber;

	/**
	 * The directory to which the data should be written. Specified relative to the
	 * ExtractRoot
	 */
	@FieldRequired
	public String ExtractionDirectory;

	/**
	 * DateTime at which the extraction request was submitted
	 */
	@FieldRequired
	public String JobSubmittedAt;

	/**
	 * True if this is an identifiable extraction (i.e. files should not be anonymised)
	 */
	@FieldRequired
    public boolean IsIdentifiableExtraction;

    /**
	 * True if this is a "no filters" (i.e. no file rejection filters should be applied)
	 */
	@FieldRequired
    public boolean IsNoFilterExtraction;

    /**
    * True if this extraction uses the global pool of DICOM files
    */
    @FieldRequired
    public boolean IsPooledExtraction;

	protected ExtractMessage() {
	}
}

File: src/common/Smi.Common/Messages/Extraction/ExtractMessage.cs
using Equ;
using Newtonsoft.Json;
using System;

namespace Smi.Common.Messages.Extraction
{
    /// <summary>
    /// Base class for all messages relating to the extract process
    /// </summary>
    public abstract class ExtractMessage : MemberwiseEquatable<ExtractMessage>, IExtractMessage
    {
        [JsonProperty(Required = Required.Always)]
        public Guid ExtractionJobIdentifier { get; set; }

        [JsonProperty(Required = Required.Always)]
        public string ProjectNumber { get; set; } = null!;

        [JsonProperty(Required = Required.Always)]
        public string ExtractionDirectory { get; set; } = null!;

        [JsonProperty(Required = Required.Always)]
        public DateTime JobSubmittedAt { get; set; }

        [JsonProperty(Required = Required.Always)]
        public bool IsIdentifiableExtraction { get; set; }

        [JsonProperty(Required = Required.Always)]
        public bool IsNoFilterExtraction { get; set; }

        [JsonConstructor]
        protected ExtractMessage() { }


        protected ExtractMessage(
            Guid extractionJobIdentifier,
            string projectNumber,
            string extractionDirectory,
            DateTime jobSubmittedAt,
            bool isIdentifiableExtraction,
            bool isNoFilterExtraction)
            : this()
        {
            ExtractionJobIdentifier = extractionJobIdentifier;
            ProjectNumber = projectNumber;
            ExtractionDirectory = extractionDirectory;
            JobSubmittedAt = jobSubmittedAt;
            IsIdentifiableExtraction = isIdentifiableExtraction;
            IsNoFilterExtraction = isNoFilterExtraction;
        }

        protected ExtractMessage(IExtractMessage request)
            : this(
                request.ExtractionJobIdentifier,
                request.ProjectNumber,
                request.ExtractionDirectory,
                request.JobSubmittedAt,
                request.IsIdentifiableExtraction,
                request.IsNoFilterExtraction)
        { }

        public override string ToString() =>
            $"ExtractionJobIdentifier={ExtractionJobIdentifier}, " +
            $"ProjectNumber={ProjectNumber}, " +
            $"ExtractionDirectory={ExtractionDirectory}, " +
            $"JobSubmittedAt={JobSubmittedAt:s}, " +
            $"IsIdentifiableExtraction={IsIdentifiableExtraction}, " +
            $"IsNoFilterExtraction={IsNoFilterExtraction}, " +
            "";
    }
}
